spec/factories/ext_management_system.rb

```diff
@@ -28,93 +28,124 @@
         create_list :storage, evaluator.storage_count, :ext_management_system => ems
       end
     end
+
+    trait :with_invalid_authentication do
+      after(:create) do |x|
+        x.authentications << FactoryGirl.build(:authentication, :resource => x, :status => "invalid")
+      end
+    end
   end
 
   # Intermediate classes
 
   factory :ems_infra,
           :aliases => ["manageiq/providers/infra_manager"],
           :class   => "ManageIQ::Providers::InfraManager",
-          :parent  => :ext_management_system
+          :parent  => :ext_management_system do
+  end
 
   factory :ems_physical_infra,
           :aliases => ["manageiq/providers/physical_infra_manager"],
           :class   => "ManageIQ::Providers::PhysicalInfraManager",
-          :parent  => :ext_management_system
+          :parent  => :ext_management_system do
+  end
+
+  factory :ems_physical_infra_with_invalid_authentication,
```

Comment:
It looks like something went wrong during a rebase and brought this factory back as well as all of the `do / end` blocks I deleted :)

Reply:
I `do` think this should be the `end` of those blocks, indeed.